### ISSUE TITLE
[FEA] Drop Duplicates

### ISSUE THREAD

**Issue by brhodes10** [rapidsai/cudf#1467]
**Is your feature request related to a problem? Please describe.**
I wish I could use cudf to drop duplicate rows.

**Describe the solution you'd like**
A function that drops the duplicate rows from a dataframe

**Describe alternatives you've considered**
groupby

**Additional context**
Add any other context, code examples, or references to existing implementations about the feature request here.
This notebook https://github.com/rapidsai/notebooks-extended/blob/master/tutorials/cyber/network_mapping/lanl_network_mapping_using_rapids.ipynb
Or more specifically this line.
https://github.com/rapidsai/notebooks-extended/blob/e0e106909a38ba9118e5b871b3bd41cd02defa5b/tutorials/cyber/network_mapping/lanl_network_mapping_using_rapids.ipynb#L501

**Comment by randerzander** [comment]
For future implementation reference, the Pandas equivalent method is called [drop_duplicates](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.drop_duplicates.html)

**Comment by harrism** [comment]
@karthikeyann can you look at implementing the libcudf part of this? I think the API should be something like: `cudf::table drop_duplicates(cudf::table const & input);` and should be defined in cudf/include/stream_compaction.hpp (this is a new file that I created in the yet-to-be-merged #1654 ).

As for implementation, note that this must preserve the order of the input, which makes it tricky.

I can think of three approaches. The naive approach is to sort the rows using `thrust::sort` with [`rows_equal`](https://github.com/rapidsai/cudf/blob/e489bf6e223795a1e0c2b4c5e10bcca1f2ff4edd/cpp/src/table/device_table.cuh#L215) as the comparator. Then use `thrust::unique()`. But then to get them back into the original order you have to sort the result by the original row index (which you have to carry along through the above)...

Better would be to compute the hash value for each row, sort using hash values as keys and input row index as values. Then unique(). Then sort just the remaining row indices and use that to gather. I'm assuming that `gdf_hash` returns a unique hash value for two rows if they contain different values...

An approach without sorting (but possibly high constant for hashing)
```
for N rows…
1. Hash the rows into a hash table in parallel using a min aggregator on the row index. This is a groupby.min.
2. Create a boolean column B of size N.
3. In parallel: B[i] = (hash_table(row_i) == i)
4. For each column: apply_boolean_mask(column, B)  (See #1654 )
```

Once I extend apply_boolean_mask to operate on tables rather than single columns, step 4 becomes faster and easier.

The question is, do we have the hash table functionality we need for 1 and 3?

Also, need to look at the Pandas API (linked [above](https://github.com/rapidsai/cudf/issues/1467#issuecomment-485299183))  to find out how it deals with nulls in the input, and make sure we match that.

CC @jrhemstad for input on hashing.

**Comment by harrism** [comment]
@randerzander: @jrhemstad points out that Pandas groupby returns a sorted result, but we don't do that in libcudf. So why do we need to match Pandas on the ordering of drop_duplicates?
